### PR TITLE
F010 appointment component

### PR DESCRIPTION
**Pull request by grzegorzbors**
Component contains links to test DB - this will be changed when the proper backend is developed. I used Formik library insted of creating the form from scratch.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -11,6 +11,7 @@
     "axios": "^0.21.1",
     "date-fns": "^2.22.1",
     "dotenv": "^9.0.2",
+    "formik": "^2.2.9",
```
Comment (mwojtow) [inline package.json:14]: I suggest to add yup validator package, to validate inputs

File: src/utils/formUtils.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    .max(20, "Dopuszczalna długość 20 liter")
+    .required("Pole wymagane"),
+  email: Yup.string().email("Niepoprawny adres").required("Pole wymagane"),
+  phone: Yup.string()
+    .matches(phoneRegExp, "Niepoprawny numer")
+    .required("Pole wymagane"),
+});
+
+export const initialValues = {
+  date: "",
```
[the remark this answers is not in the document]
Comment (grzegorzbors) [reply src/utils/formUtils.js:24]: Assigned today's date using date-fns.

File: src/components/AppointmentForm/FormikControl.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  switch (control) {
+    case "text":
+      return <Input {...rest} />;
+    case "email":
+      return <Input {...rest} />;
+    case "select":
+      return <Select {...rest} />;
+    case "date":
+      return <Input {...rest} />;
+    case "time":
```
[the remark this answers is not in the document]
Comment (grzegorzbors) [reply src/components/AppointmentForm/FormikControl.js:15]: FormField is changing dynamically based on props based to it, so it also covers date. It is also not quite set in stone how the future date picker will work - it probably will be a select element that pulls data from server and shows available time to pick. In the meantime I will do a research on libs relating to time :)

File: src/components/AppointmentForm/NewAppointmentForm.js [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+
+  return (
+    <Formik
+      initialValues={initialValues}
+      validationSchema={validate}
+      onSubmit={onSubmitHandler}
+    >
+      {({ handleChange, isSubmitting }) => (
+        <Form className={appointmentFormStyles.newAppointmentForm}>
+          <FormikControl control="date" label="Data" type="date" name="date" />
```
[the remark this answers is not in the document]
Comment (grzegorzbors) [reply src/components/AppointmentForm/NewAppointmentForm.js:65]: Extracted separate file with objects array and mapped over it inside the form.

File: src/components/AppointmentForm/NewAppointmentForm.js [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+      setDoctorsList(response.data);
+    } catch {
+      setIsError(true);
+    }
+  };
+
+  const onSubmitHandler = async (values, { setSubmitting }) => {
+    try {
+      setSubmitting(true);
+      await axios({
```
[the remark this answers is not in the document]
Comment (grzegorzbors) [reply src/components/AppointmentForm/NewAppointmentForm.js:39]: Extracted to _api/requests.js_ as suggested.

File: src/App.css [hunk 6]
```diff
@@ -1,5 +1,7 @@
+html {
+  min-height: 100vh;
```
Comment (mwojtow) [inline src/App.css:2]: I think that is usually in body tag

File: src/components/AppointmentForm/formikControlsCfg.js [hunk 7]
```diff
@@ -0,0 +1,20 @@
+const formikControls = [
```
Comment (mwojtow) [inline src/components/AppointmentForm/formikControlsCfg.js:1]: short name for file/const could just be `formConfig` or more specific `appointmentsFormConfig`, and it's better to have full name, even for obvious cfg = config - don't worry it's long, it's more readable

File: src/utils/formValidationUtils.js [hunk 8]
```diff
@@ -0,0 +1,33 @@
+import * as Yup from "yup";
+import format from "date-fns/format";
+
+const phoneRegExp =
+  /^((\\+[1-9]{1,4}[ \\-]*)|(\\([0-9]{2,3}\\)[ \\-]*)|([0-9]{2,4})[ \\-]*)*?[0-9]{3,4}?[ \\-]*[0-9]{3,4}?$/;
+
+export const appointmentValidation = Yup.object({
+  date: Yup.string().required("Data wymagana"),
```
Comment (mwojtow) [inline src/utils/formValidationUtils.js:8]: More generic info: "This field is required", you can reuse it

File: src/components/AppointmentForm/NewAppointmentForm.js [hunk 9]
```diff
@@ -54,78 +62,48 @@ const NewAppointmentForm = () => {
...
+            <Button
+              type="submit"
+              variant="contained"
+              color="primary"
+              disabled={isSubmitting}
+              className={appointmentFormStyles.submitButton}
+            >
+              Umów
+            </Button>
+            {isError && (
```
Comment (mwojtow) [inline src/components/AppointmentForm/NewAppointmentForm.js:96]: you could move that error to toast notifier

File: src/components/AppointmentForm/NewAppointmentForm.js [hunk 10]
```diff
@@ -54,78 +62,48 @@ const NewAppointmentForm = () => {
     fetchDoctorsList();
   }, []);
 
+  const notify = () => toast.success("Wizyta umówiona!");
```
Comment (mwojtow) [inline src/components/AppointmentForm/NewAppointmentForm.js:65]: could be extracted out of component

**Review by mwojtow — APPROVED** [review]

**Pull request merged**